src/sinks/prometheus/exporter.rs

```diff
@@ -369,6 +404,60 @@ mod integration_tests {
         assert_eq!(data["value"][1], Value::String("123.4".into()));
     }
 
+    async fn reset_on_flush_preiod() {
+        let config = PrometheusExporterConfig {
+            address: PROMETHEUS_ADDRESS.parse().unwrap(),
+            flush_period_secs: 3,
+            ..Default::default()
+        };
+        let (sink, _) = config.build(SinkContext::new_test()).await.unwrap();
+        let (tx, rx) = mpsc::unbounded_channel();
+        tokio::spawn(sink.run(Box::pin(rx)));
+
+        let (name1, event) = create_metric_set(None, vec!["0", "1", "2"]);
+        tx.send(event).expect("Failed to send.");
+        let (name2, event) = create_metric_set(None, vec!["3", "4", "5"]);
+        tx.send(event).expect("Failed to send.");
+
+        // Wait a bit for the prometheus server to scrape the metrics
+        time::delay_for(time::Duration::from_secs(2)).await;
+
+        // Now try to download them from prometheus
+        let result = prometheus_query(&name1).await;
+        assert_eq!(
+            result["data"]["result"][0]["value"][1],
+            Value::String("3".into())
+        );
+        let result = prometheus_query(&name2).await;
+        assert_eq!(
+            result["data"]["result"][0]["value"][1],
+            Value::String("3".into())
+        );
+
+        // Wait a bit for expired metrics
+        time::delay_for(time::Duration::from_secs(2)).await;
+
+        let (name1, event) = create_metric_set(Some(name1), vec!["6", "7"]);
+        tx.send(event).expect("Failed to send.");
+        let (name2, event) = create_metric_set(Some(name2), vec!["8", "9"]);
+        tx.send(event).expect("Failed to send.");
+
+        // Wait a bit for the prometheus server to scrape the metrics
+        time::delay_for(time::Duration::from_secs(2)).await;
```

Comment:
Is this enough time to *guarantee* they are expired with a 3 second expiry?

Reply:
I think yes, this should be enough. With **2** seconds above we have **4** in total. I'll add **1** more second to be sure that we are ok :slightly_smiling_face: